include/swift/SIL/SILBridgingImpl.h

```diff
@@ -1813,14 +1819,68 @@ BridgedDeclRef BridgedWitnessTableEntry::getMethodRequirement() const {
+OptionalBridgedFunction BridgedWitnessTableEntry::getMethodWitness() const {
   return {unbridged().getMethodWitness().Witness};
 }
 
+BridgedAssociatedTypeDecl BridgedWitnessTableEntry::getAssociatedTypeRequirement() const {
+  return {unbridged().getAssociatedTypeWitness().Requirement};
+}
+
+BridgedType BridgedWitnessTableEntry::getAssociatedTypeWitness() const {
+  return swift::SILType::getPrimitiveObjectType(unbridged().getAssociatedTypeWitness().Witness);
+}
+
+BridgedType BridgedWitnessTableEntry::getAssociatedTypeProtocolRequirement() const {
+  return swift::SILType::getPrimitiveObjectType(unbridged().getAssociatedTypeProtocolWitness().Requirement);
+}
+
+BridgedProtocolDecl BridgedWitnessTableEntry::getAssociatedTypeProtocolDecl() const {
+  return {unbridged().getAssociatedTypeProtocolWitness().Protocol};
+}
+
+BridgedProtocolConformance BridgedWitnessTableEntry::getAssociatedTypeProtocolWitness() const {
+  return {unbridged().getAssociatedTypeProtocolWitness().Witness};
+}
+
+BridgedProtocolDecl BridgedWitnessTableEntry::getBaseProtocolRequirement() const {
+  return {unbridged().getBaseProtocolWitness().Requirement};
+}
+
+BridgedProtocolConformance BridgedWitnessTableEntry::getBaseProtocolWitness() const {
+  return swift::ProtocolConformanceRef(unbridged().getBaseProtocolWitness().Witness);
+}
+
+
+BridgedWitnessTableEntry BridgedWitnessTableEntry::createInvalid() {
+  return swift::SILWitnessTable::Entry();
+}
+
 BridgedWitnessTableEntry BridgedWitnessTableEntry::createMethod(BridgedDeclRef requirement,
-                                                                BridgedFunction function) {
-  return swift::SILWitnessTable::Entry(swift::SILWitnessTable::MethodWitness{requirement.unbridged(),
-                                                                             function.getFunction()});
+                                                                OptionalBridgedFunction witness) {
+  return swift::SILWitnessTable::Entry(
+    swift::SILWitnessTable::MethodWitness{requirement.unbridged(), witness.getFunction()});
+}
+
+BridgedWitnessTableEntry BridgedWitnessTableEntry::createAssociatedType(BridgedAssociatedTypeDecl requirement,
+                                                                        BridgedType witness) {
+  return swift::SILWitnessTable::Entry(
+    swift::SILWitnessTable::AssociatedTypeWitness{requirement.unbridged(), witness.unbridged().getASTType()});
+}
+
+BridgedWitnessTableEntry BridgedWitnessTableEntry::createAssociatedTypeProtocol(BridgedType requirement,
+                                                                                BridgedProtocolDecl protocolDecl,
+                                                                                BridgedProtocolConformance witness) {
+  return swift::SILWitnessTable::Entry(
+    swift::SILWitnessTable::AssociatedTypeProtocolWitness{requirement.unbridged().getASTType(),
```

Comment:
Ok, I see now that SILWitnessTable calls it "associated type protocol witness" too. But let's change it to AssociatedConformanceWitness